### ISSUE TITLE
Implement unforked Flutter compilation

### ISSUE THREAD

**Issue by RedBrogdon** [dart-lang/dart-pad#1282]
This is a big change, so this may end up functioning as a meta-issue.

**Comment by johnsonmh** [comment]
Related to doing the un-fork, what version of Flutter will DartPad use? Latest release? Dev? 

Probably unrealistic, but it would be really cool to allow for specifying which version of Flutter you point to. And it would be *mega* cool if you could even point it to un-merged versions of Flutter. That way we could link to demos in our PRs. This would also help us when we're generating local docs, since the dartdoc embedder could properly render an up-to-date sample.

**Comment by legalcodes** [comment]
Hope it's alright if I re-post here (after closing #1273, happy to track in a separate issue if needed).

This fix also solves the following bug:

TextField isn't working properly in package:flutter_web, see, e.g.:
https://dartpad.dev/embed-flutter.html?id=ac6b3d73b86f584da1f86d1e8574b9ac

**Comment by RedBrogdon** [comment]
@legalcodes Totally alright. If nothing else, it reminds us to check that your problem gets fixed.

@johnsonmh There's been an open FR for multiple SDK version for a while at #774, and it's something that should see traction before the end of the year. Support for unmerged (or even bleeding edge master) versions of the SDK isn't on our roadmap, though. There'd be a ton of work to make that happen (in terms of dev time and CPU load, tbh!).

**Comment by johnsonmh** [comment]
A side effect of using an older `flutter_web` version is that the version being used has a `TextField` bug, so the recently added `TextField` dartpad sample has a bug in it (text can not be input into the demo).

Text field demo: https://master-api.flutter.dev/flutter/material/TextField-class.html#material.TextField.2

If the `flutter_web` version used by dartpad isn't going to be updated soon, we should consider changing the `dartpad` to a `snippet` in the `TextField` documentation for now.

cc @justinmc

**Comment by domesticmouse** [comment]
I'm in the process of implementing this in https://github.com/dart-lang/dart-services/pull/452 but I'm blocked on https://github.com/dart-lang/sdk/issues/38702.

I'm hopeful we can get this in next week, but there are unknown unknowns in the mix, as well as migrating the already deployed sample code to the new required formats (updating the imports at a minumum) that may push out migration further.

**Comment by justinmc** [comment]
Sorry I missed this mention.  It looks like https://github.com/dart-lang/sdk/issues/38702 was just reopened, so if it's actually going to take a lot longer now then I'm fine with changing it to a `snippet`.  Otherwise it's probably fine if it won't go to the stable docs for awhile.

**Comment by domesticmouse** [comment]
I've closed that issue again. The mistake I was making w.r.t. dart-lang/sdk#38702 was not running `flutter precache --web` on changing versions, and thus winding up with mismatched dill files for the compiler I was running. With that said, I'm sure there are a handful of other gotchas waiting for me as I push this forward.

**Comment by RedBrogdon** [comment]
Closed by https://github.com/dart-lang/dart-services/pull/467
